api/v1/views/cities.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+from api.v1.views import app_views
+from models import storage, City
+from flask import abort, request, jsonify
+
+@app_views.route('/states/<state_id>/cities', strict_slashes=False, methods=['GET'])
+def city_by_state_id(state_id):
+    """
+    Retrieves list of all City objects of a State
+    """
+    cities = storage.all("City").values()
+    city_list = [city.to_dict() for city in cities if city.state_id == state_id]
+    if len(city_list) == 0:
+            abort(404)
+    return jsonify(city_list)
+
+@app_views.route('/cities/<city_id>', strict_slashes=False, methods=['GET'])
+def city_by_id(city_id):
+    """
+    Retrieves city by city id. If city_id not linked to City,
+    raise 404.
+    """
+    cities = storage.all("City").values()
+    city = next(filter(lambda x: x.id == city_id, cities), None)
+    return jsonify(city.to_dict()) if city else abort(404), 201
+
+@app_views.route('/cities/<city_id>', strict_slashes=False, methods=['DELETE'])
+def delete_city(city_id):
+    """
+    Deletes city by id. If city_id not linked to City, raise 404
+    Returns empty dict with status 200
+    """
+    cities = storage.all("City").values()
+    city = next(filter(lambda x: x.id == city_id, cities), None)
+    if city:
+        storage.delete(city)
+        return jsonify({}), 200
+    else:
+        abort(404)
+
+@app_views.route('/states/<state_id>/cities', strict_slashes=False, methods=['POST'])
```

Comment:
This line will give us pain because it is longer than 80 symbols, so it will not pass the PEP8 check